### PR TITLE
Extended datatypes

### PR DESCRIPTION
**Pull request by joedevivo**
This one's for you @evanmcc and @jtuple 

Now, when writing a schema, you can add multiple datatypes. Here's how it works:

In a mapping, `{datatype, Type}` still works as it used to, but if you want, you can replace Type with a List of types.

Example: anti_entropy.expire

``` erlang
{mapping, "anti_entropy.expire", "riak_kv.anti_entropy.expire", [
    {datatype, [{duration, ms}, {atom, never}]}
]}.
```

now if you set something like `anti_entropy.expire = right_away` it will try converting "right_away" into a duration first. When that doesn't work, it'll try converting it to an atom and then compare to 'never'. Only if these all fail will it produce an error, and the first one to pass is the one it uses.

You can put any datatype that currently exists, and I've added a few more. These extended types are single instances of a type, like `{atom, never}`, which only allows the atom 'never'.

I hope that makes sense. Better documentation for this will come next week on the cuttlefish wiki.

### CODE REVIEW

File: src/cuttlefish_generator.erl [hunk 1]
```diff
@@ -364,26 +364,84 @@ transform_datatypes(Conf, Mappings) ->
...
-                                true -> {[{Variable, NewValue}|Acc], ErrorAcc};
-                                false ->
-                                    ErrStr = io_lib:format("Bad value: ~s for enum ~s", [NewValue, Variable]),
-                                    lager:error(ErrStr),
-                                    {Acc, [{error, ErrStr}|ErrorAcc]}
-                            end;
-                        {_, NewValue} -> {[{Variable, NewValue}|Acc], ErrorAcc}
+                    DTs = cuttlefish_mapping:datatype(MappingRecord),
+
+                    %% DTs is a list now!
```
Comment (seancribbs) [inline src/cuttlefish_generator.erl:369]: What if it's not? Will old schemas still work?

Comment (seancribbs) [inline src/cuttlefish_generator.erl:369]: Oh, I see, the parsing handles it.

**Comment by seancribbs** [comment]
Codez seem fine, nice unit tests too. @evanmcc has promised to try it out on a schema.

**Comment by evanmcc** [comment]
testing with `fsm_limit = 50000 | undefined` works as expected.  will drop a pr for that today.

fwiw I did get this warning on startup:

```
12:35:27.563 [warning] cuttlefish_util:conf_get_value/2 has been deprecated. use cuttlefish:conf_get/2
config is OK
```

otherwise, works great, thanks!

**Comment by joedevivo** [comment]
@evanmcc you might be working with an older schema. I replaced all those function calls this week in each repo.

**Comment by seancribbs** [comment]
:+1: merge away

**Pull request merged**